### PR TITLE
[PactService] Truly provide sensible values for (chain-data) on local calls

### PR DESCRIPTION
**Pull request by emmanueldenloye**
I added an  extra test to `ChainData.hs` to cover the local case.

### CODE REVIEW

File: src/Chainweb/Pact/PactService.hs [hunk 1]
```diff
@@ -796,9 +796,9 @@ execLocal cmd = withDiscardedBatch $ do
     -- necessary information for this call to return sensible values.
     parentHeader <- liftIO $! lookupM bhDb bhash
     withCheckpointer target "execLocal" $ \(PactDbEnv' pdbenv) -> do
-        PactServiceEnv{..} <- ask
         mc <- use psInitCache
-        r <- withBlockData parentHeader $
+        r <- withBlockData parentHeader $ do
+             PactServiceEnv{..} <- ask
```
Comment (slpopejoy) [inline src/Chainweb/Pact/PactService.hs]: This seems like an unnecessary change -- if it is let's remove it to keep this PR focused on tests

Comment (slpopejoy) [inline src/Chainweb/Pact/PactService.hs]: Is this the fix?

Comment (emmanueldenloye) [reply src/Chainweb/Pact/PactService.hs]: Yes, this is the fix.
